docs/learn/WSL及docker的安装 .md

```diff
@@ -0,0 +1,188 @@
+WSL2正好相反  
+虚拟机需要配置共享文件夹
+
+4. 兼容性平衡
+
+WSL1对Windows兼容更好，但部分Linux功能不支持
+
+WSL2几乎完整支持Linux内核功能，但对GPU/USB等外设支持有限
+
+传统虚拟机能实现完整硬件支持
+
+开发选WSL，生产环境用虚拟机，日常轻度使用WSL2体验最佳。两者可以共存，根据需求切换使用。
+
+### 二.WSL的安装及安装时可能有的报错
+1.必要条件
+
+<font style="color:rgb(25, 27, 31);">Windows 10 2020年5月(2004) 版, Windows 10 2019年5月(1903) 版，或者 Windows 10 2019年11月(1909) 版</font>
+
+WSL 2 使用了 Hyper-V 架构的一部分功能，但对 Windows 11 的版本并没有限制。家庭版、教育版、专业版和企业版都可以安装。
+
+确保你的计算机是最新版本
+
+2.需要配置的环境和工具（win10下）
+
+首先确保电脑打开了**适用于linux的windows子系统**和**虚拟机平台**{
+
+win+i打开**设置**，搜索**可选功能**，下划找到**更多windows功能**
+
+在里边找到适用于linux的windows子系统和虚拟机平台}
+
+微软应用商店下载**终端**（推荐使用）
+
+还得有把~~梯子~~
+
+3.安装WSL
+
+**启用WSL**
+
+以管理员身份运行终端
+
+`dism.exe /online /enable-feature /featurename:Microsoft-Windows-Subsystem-Linux /all /norestart`
+
+
+
+此步可能报错：<font style="color:rgb(79, 79, 79);">无法解析服务器的名称或地址</font>
+
+<font style="color:rgb(77, 77, 77);">解决办法是修改域名DNS</font>
+
+按`<font style="background-color:#FFFFFF;">windows + i</font>`快捷键打开设置，在左侧菜单栏找到网络和internet
+
+<font style="color:rgba(0, 0, 0, 0.75);">点击高级网络设置</font>
+
+<font style="color:rgba(0, 0, 0, 0.75);">选中当前使用的网络，点击查看其它属性，之后点击编辑</font>
+
+<font style="color:rgba(0, 0, 0, 0.75);">修改dns，将自动代理改为手动IPv4，输入114.114.114.114</font>
+
+<font style="color:rgba(0, 0, 0, 0.75);"></font>
+
+**<font style="color:rgba(0, 0, 0, 0.75);">启用虚拟机平台</font>**
+
+<font style="color:rgb(25, 27, 31);">在 Windows 10（2004）上启用虚拟机平台</font>
+
+`dism.exe /online /enable-feature /featurename:VirtualMachinePlatform /all /norestart`
+
+<font style="color:rgb(25, 27, 31);">在 Windows 10（1903，1909）上启用虚拟机平台</font>
+
+`Enable-WindowsOptionalFeature -Online -FeatureName VirtualMachinePlatform -NoRestart`
+
+
+
+可能报错：0x800701bc
+
+终端输入wsl --update即可，更新一下wsl
+
+可能报错：wsl更新进度为一直为0或者不变
+
+`wsl --install --web-download`
+
+让他从github下载，或者重启一下试试
+
+
+
+**<font style="color:rgb(25, 27, 31);">设置 WSL 2 为默认值</font>**
+
+`wsl --set-default-version 2`
+
+
+
+**<font style="color:rgb(25, 27, 31);">安装一个 Linux 发行版</font>**
+
+<font style="color:rgb(25, 27, 31);">微软应用商店搜索ubuntu20.04下载即可</font>
+
+<font style="color:rgb(25, 27, 31);">下载好后打开设置用户名密码，WSL的安装到这里就结束了</font>
+
+<font style="color:rgb(25, 27, 31);"></font>
+
+### <font style="color:rgb(25, 27, 31);">三.安装docker及安装时可能有的报错</font>
+建议装的时候全程开代理TUN模式
+
+**首先卸载可能会冲突的软件包**
+
+打开下载的ubuntu后输入
+
+`for pkg in docker.io docker-doc docker-compose docker-compose-v2 podman-docker containerd runc;do sudo apt-get remove $pkg;done`
+
+
+
+**使用存储库安装apt**
+
+```powershell
+# Add Docker's official GPG key
+sudo apt-get update
+sudo apt-get install -y ca-certificates curl
+sudo install -m 0755 -d /etc/apt/keyrings
+sudo curl -fsSL https://mirrors.openctf.dev/https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
+sudo chmod a+r /etc/apt/keyrings/docker.asc
+
+# Add the repository to Apt sources
+echo \
+  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://mirrors.openctf.dev/https://download.docker.com/linux/ubuntu \
```

Comment:
newstar do not provide this mirror service. if ok please refer to docker official document.